app/controllers/admin/badge_achievements_controller.rb

```diff
@@ -0,0 +1,49 @@
+module Admin
+  class BadgeAchievementsController < Admin::ApplicationController
+    layout "admin"
+
+    def index
+      @q = BadgeAchievement
+        .includes(:badge)
+        .includes(:user)
+        .order(created_at: :desc)
+        .ransack(params[:q])
+      @badge_achievements = @q.result.page(params[:page] || 1).per(15)
+    end
+
+    def destroy
+      @badge_achievement = BadgeAchievement.find(params[:id])
+
+      if @badge_achievement.destroy
+        flash[:success] = "Badge has been deleted!"
```

Comment:
Nitpick: Could this be changed to "Badge achievement has been deleted!"?

Reply:
Great call @Zhao-Andy ! I've made this change.